src/client/hooks/useFetch.js

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+import { getUserFirebaseToken } from '../firebase/auth';
+/**
+ * Fetch API
+ * https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
+ *
+ * Wrapper of fetch to give some extra capabilities like:
+ * - Loading
+ * - Error handling
+ *
+ * And main purpose to deal with Auth headers so it is abstracted.
+ * As we use the firebase token for authN/Z
+ */
+const useFetch = (url, options) => {
+  const [response, setResponse] = useState(null);
+  const [error, setError] = useState(null);
+  const [loading, setLoading] = useState(false);
+  useEffect(() => {
+    const doFetch = async () => {
```

Comment:
nit: maybe this function can be extracted outside of the hook, just in case if someone wants to do fetch from outside react (event handlers, fire and forget and so on)